### ISSUE TITLE
Why is PPO training so different when running with SB3 image normalization compared to custom vectorized image scaler wrapper?

### ISSUE THREAD

**Issue by victorsevero** [DLR-RM/stable-baselines3#1863]
### 🐛 Bug

I'm currently working on a custom environment made with [stable-retro](https://github.com/Farama-Foundation/stable-retro). I'm using PPO with `CnnPolicy`. These are the observation spaces during each relevant step of environment wrapping:

* Before any wrapper (single RGB image): `Box(0, 255, (224, 240, 3), uint8)`
* After WarpFrame (last single-process wrapper): `Box(0, 255, (84, 84, 1), uint8)`
* After vectorized wrappers (`VecFrameStack`, `VecTransposeImage`): `Box(0, 255, (2, 84, 84), uint8)`

With that setup, I leave the default parameters for `policy_kwargs`:

```python
PPO(
    ...,  # other parameters
    policy_kwargs=None,
)
```

Which is equivalent to setting `normalize_images=True` according to [here](https://github.com/DLR-RM/stable-baselines3/blob/master/stable_baselines3/common/policies.py#L812)

I set a fixed seed; multiple runs with same parameters are consistent with each other.

Then I made a custom `VecImageScaling` wrapper set after the previous last vectorized wrapper

```python
class VecImageScaling(VecEnvWrapper):
    def __init__(self, venv: VecEnv):
        observation_space = spaces.Box(
            0.0,
            1.0,
            shape=venv.observation_space.shape,
            dtype=np.float32,
        )
        super().__init__(venv, observation_space=observation_space)

    def step_wait(self):
        observations, rewards, dones, infos = self.venv.step_wait()
        return observations / 255, rewards, dones, infos

    def reset(self):
        observations = self.venv.reset()
        return observations / 255
```

After this one, this is the observation space: `Box(0.0, 1.0, (2, 84, 84), float32)`. While using this wrapper, I disabled the `normalize_images` parameter with

```python
PPO(
    ...,  # other parameters
    policy_kwargs={"normalize_images": False},
)
```

the results are significantly different, as you can see in the following images.

![image](https://github.com/DLR-RM/stable-baselines3/assets/7810753/5ad530b7-589c-4ab6-ae80-c00521233d20)
![image](https://github.com/DLR-RM/stable-baselines3/assets/7810753/60363ad6-e518-4090-a015-2440cd3f1d32)

Multiple runs with same parameters on this setup are still consistent with each other (exactly same results).

Why are they so different if, theoretically, I'm applying the same preprocessing step? Does SB3 preprocessing does something else? I looked at SB3 codebase and the [internal normalization](https://github.com/DLR-RM/stable-baselines3/blob/master/stable_baselines3/common/preprocessing.py#L120
) seems to be exactly the same as the one I made, the only difference being that SB3 makes that in `pytorch` instead of `numpy`. But there's no randomness in this operation and they even have the same target float precision (`torch.Tensor.float()` is an alias for `torch.Tensor.to(torch.float32)` and we can see that the observation space after the last wrapper is `float32` too).

Thanks!

### System Info

- OS: Linux-5.15.133.1-microsoft-standard-WSL2-x86_64-with-glibc2.35 # 1 SMP Thu Oct 5 21:02:42 UTC 2023
- Python: 3.10.13
- Stable-Baselines3: 2.3.0a2
- PyTorch: 2.2.1+cu121
- GPU Enabled: True
- Numpy: 1.26.4
- Cloudpickle: 3.0.0
- Gymnasium: 0.29.1

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] I have provided a [minimal and working](https://github.com/DLR-RM/stable-baselines3/issues/982#issuecomment-1197044014) example to reproduce the bug
- [X] I have checked my env using the env checker
- [X] I've used the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by araffin** [comment]
Hello,
for a bit more context, which one is which in the picture?

**Comment by araffin** [comment]
Using your wrapper, I'm obtaining the exact same output:
```python
import numpy as np
import torch as th
from gymnasium import spaces
from stable_baselines3 import PPO
from stable_baselines3.common.env_util import make_atari_env
from stable_baselines3.common.vec_env import VecEnv, VecEnvWrapper, VecTransposeImage


class VecImageScaling(VecEnvWrapper):
    def __init__(self, venv: VecEnv):
        observation_space = spaces.Box(
            0.0, 1.0, shape=venv.observation_space.shape, dtype=np.float32
        )
        super().__init__(venv, observation_space=observation_space)

    def step_wait(self):
        observations, rewards, dones, infos = self.venv.step_wait()
        return observations / 255.0, rewards, dones, infos

    def reset(self):
        observations = self.venv.reset()
        return observations / 255.0


vec_env = make_atari_env("PongNoFrameskip-v4", n_envs=1, seed=0)

# With the wrapper
vec_env = VecTransposeImage(vec_env)
vec_env = VecImageScaling(vec_env)
model = PPO(
    "CnnPolicy", vec_env, verbose=1, seed=0, policy_kwargs=dict(normalize_images=False)
)

obs = vec_env.reset()
obs_tensor, _ = model.policy.obs_to_tensor(obs)
values_with_wrapper = model.policy.predict_values(obs_tensor)

# Without the wrapper
vec_env = make_atari_env("PongNoFrameskip-v4", n_envs=1, seed=0)
vec_env = VecTransposeImage(vec_env)
model = PPO(
    "CnnPolicy", vec_env, verbose=1, seed=0, policy_kwargs=dict(normalize_images=True)
)

obs = vec_env.reset()
obs_tensor, _ = model.policy.obs_to_tensor(obs)
values_without_wrapper = model.policy.predict_values(obs_tensor)


assert th.allclose(values_with_wrapper, values_without_wrapper)
```
